README.md

```diff
@@ -1,9 +1,71 @@
 # terraform-scripts
 
 This repository holds our common Terraform Open edX deployment scripts.
+They can be used to provision and manage the infrastructure required by an Open edX instance on AWS.
 
-Currently it actually contains two different ways to provision AWS resources
+### TODOS:
+- [ ] Write usage documentation
+- [ ] Expand the conventions section below with more details and examples
+
+## Architecture map
+
+See [this document](https://docs.google.com/drawings/d/1s2hd6hJSKo1a-eqp6rI2kF-afACZXlK3f3ymptZjV6M/edit?usp=sharing).
+
+## Conventions
+### Resource naming
+
+Every resource provisioned by Terraform should be created using the following naming schema:
+
+```
+`{client_shortname}-{environment}-{resource}-{additional_info}`
+```
+
+Examples:
+- `opencraftx-production-appserver-1` - Appserver from the production environment of client `OpenCraftX`
+- `example-stage-alb` - Alb resource for client `Example`
+
+To ease the migration and import process, the resources were renaming implies on changing credentials
+or replacement, a name override variable is available.
+
+The variables that allow name overrides are detailed on each module, and should be named as following:
+```
+override_{resource_name}_name
+```
 
+### Module categories & repo folder structure
+
+The Terraform modules must be categorized in one of the following categories:
+
+- `Core`: used to provision a vanilla Open edX instance and it's **required** dependencies, such
+as S3 buckets, databases, load balancer, security groups, etc.
```

Comment:
I think we should list here that the core modules will be the "openedx" and "director", we can then decide if we want independent directors for each "openedx" deployment or just one for all of them

Reply:
@eLRuLL I still think that we should not share director instances between two environments (`stage` and `prod` for example) for both security reasons and complexities in sharing a resource in two separate terraform environments. If this is the case, there's no reason for a director to be a standalone module.

@bradenmacdonald Do you have any inputs on this?

Reply:
That makes sense to me, I think the `openedx` module can just provision a director instance.

Reply:
Well, there could also be more environments, and for example analytics as well so the number of unused instances (directors) could increase in that case. I would maybe suggest then to still separate the director as an internal module (reused by openedx and analytics for example) provision it and setup the ec2 instance to be stopped by default so we don't unnecessary resources for the client (we would have to manually start it when we want to use them)

Reply:
@eLRuLL Since the `analytics` module is an optional module, it should use the `core` director for that environment. E.g.: production analytics should use the production director. Is there any specific reason to have a separate director instance in this case?

Reply:
right, I was thinking as analytics as its own environment as well, but right I think it should be considered as an addon for an environment